### DISCUSSION TITLE
[Question]: Future of LibreChat

### DISCUSSION THREAD

**Discussion by yoHasse** [danny-avila/LibreChat#944] [Q&A] [answered]
### Contact Details

marcus.natteldal@ltu.se

### What is your question?

I am keen on offering LibreChat as a service to our employees. 

* What does the future hold for LibreChat?
* Is LibreChat primarily a hobby project, or are there plans for it to evolve further?
* When is version 1.0 expected to be released?
* After the release of 1.0, are there plans to offer or sell support agreements?

### More Details

I serve as a solution architect at [Luleå University of Technology](https://www.ltu.se/?l=en). I've been exploring options to provide an experience similar to ChatGPT, but with the flexibility of using our Azure OpenAI service. This is due to stringent IT security regulations we adhere to.

I have tested LibreChat and recognize its potential. We are exploring the possibility of offering it as a service to both our university employees and students.

I know the questions might seem odd, but they're essential for us to make a decision about moving forward with a specific solution.

### What is the main subject of your question?

Other

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Thanks for your question! I can answer this soon when I get a minute to sit down.

**Comment by danny-avila** [comment, marked as answer]
Thanks for trying out LibreChat and I'm glad it can potentially serve your need!

I would like to answer your questions as best as I can. Before I ever learned to code, it was my dream to make a platform that would be beneficial for learning and educational orgs, and I'm honored by your consideration along with the others already using it!

> What does the future hold for LibreChat?

I hope to continue improving it; adding retrieval augmented generation (chatting with files) is a big goal I have, along with an admin dashboard, and improving the plugins implementation.

> Is LibreChat primarily a hobby project, or are there plans for it to evolve further?

It started as a hobby project, but now I like to think of it as a community project. It becomes what the community shapes it to be. I aim to steer the ship for the betterment of everyone using it.

> When is version 1.0 expected to be released?

This is hard to estimate. While I've been steady with my contributions, there is no hard business behind it and I tackle it in my free time. Contributions from others can vary in size, scope, frequency. Some of these can create backlog items, too, so I've had more scrutiny lately. This might be more related to the next questions as I come to terms with the reality of your need, and that it's already being served by others capitalizing on LibreChat within their own ventures.

My optimistic goal is to have RAG + Admin dashboard done within the next 6 months, and I would pivot towards releasing a v1.0.0 at that point. 

> After the release of 1.0, are there plans to offer or sell support agreements?

As others are already doing this exactly, I would like to offer my own services even now, as I continue to iterate on it. If this were successful, I could take it on more as a business and hire a team, while still adhering to my OSS principles (keeping it open), and the speed of a team would accelerate the developer timeline. Even being able to dedicate more time to it alone would greatly accelerate the timeline.

Let me know what you think, and you can reach out to me in any of the channels on my profile. 

I'm easily accessible on discord as well, just tag or DM danny: https://discord.librechat.ai/

You can also feel free to email me at messagedaniel@pm.me

**Reply by yoHasse** [reply]
Thank you so much for your thorough response. I truly appreciate the clarity you've provided regarding its development and future.

I'll be presenting your insights to our internal architectural council next week for further discussion. I am personally keen on exploring solutions that can integrate LibreChat, given its promising potential. Regardless of the outcome, I'll keep you informed on any of the suggested channels.

Thanks again for your time!
